src/component/Task/components/CompletedTask.js

```diff
@@ -14,7 +14,8 @@ export default function CompletedTask({ tasks, setTasks, handleDeleteTask, selec
         loadedTasks.forEach(task => newTaskMap.set(task.id, task));
 
         setTasks(newTaskMap);
-    }, []);
+    }, [selectedDate, setTasks]);
```

Comment:
의존성 배열에 setState 함수를 추가하셨는데, setState 함수는 컴포넌트가 리렌더링되더라도 동일한 함수가 계속해서 사용되기 때문에,  의존성 배열에 넣을 필요가 없다고 합니다!!
 setState를 의존성 배열에 추가하더라도 useEffect가  불필요하게 실행되지는 않기 떄문에,  성능에 문제를 일으키지는 않지만, setState가 의존성 배열에 들어가 있지 않아도, 해당 useEffect가 올바르게 동작할 것이기 때문에, setTasks를 의존성 배열에서 제거하셔도 무방할 것 같습니다!

이와 관련해서 최근 버전의 공식문서에는
'The set function has a stable identity, so you will often see it omitted from effect dependencies, but including it will not cause the effect to fire. If the linter lets you omit a dependency without errors, it is safe to do. '라고,
https://ko.react.dev/reference/react/useState
리엑트 18버전의 공식문서에는
'React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.'라고 언급되어있네요!! 한번 확인해보시면 좋을 것 같아요!!

https://legacy.reactjs.org/docs/hooks-reference.html#usestate